src/opm/parser/eclipse/EclipseState/Grid/EclipseGrid.cpp

```diff
@@ -331,18 +553,441 @@ namespace Opm {
+
+        const std::array<int, 3> dims = getNXYZ();
+        int nCells = getCartesianSize();
+
+        std::vector<double> X(8, 0.0);
+        std::vector<double> Y(8, 0.0);
+        std::vector<double> Z(8, 0.0);
+
+        m_volume.clear();
+        m_cellCenter.clear();
+        m_dx.clear();
+        m_dy.clear();
+        m_dz.clear();
+        m_depth.clear();
+
+        
+        m_volume.resize(nCells);
+        m_cellCenter.resize(nCells);
+        m_dx.resize(nCells);
+        m_dy.resize(nCells);
+        m_dz.resize(nCells);
+        m_depth.resize(nCells);
+
+        std::array<int, 3> ijk;
+        size_t n = 0;
+        
+        for (size_t k = 0; k< getNZ(); k++) {
+            for (size_t j = 0; j< getNY(); j++) {
+                for (size_t i = 0; i< getNX(); i++) {
+
+                    ijk[0] = i;
+                    ijk[1] = j;
+                    ijk[2] = k;
+
+                    getCellCorners(ijk, dims, X, Y, Z );
+
+                    m_volume[n] = calculateCellVol(X, Y, Z);
+
+                    std::array<double, 3> center;
+
+                    center[0]=std::accumulate(X.begin(), X.end(), 0.0) / 8.0;
+                    center[1]=std::accumulate(Y.begin(), Y.end(), 0.0) / 8.0;
+                    center[2]=std::accumulate(Z.begin(), Z.end(), 0.0) / 8.0;
+
+                    m_cellCenter[n] = center;
+
+                    // calculate dx
+
+                    double x2 = (X[1]+X[3]+X[5]+X[7])/4.0;
+                    double y2 = (Y[1]+Y[3]+Y[5]+Y[7])/4.0;
+
+                    double x1 = (X[0]+X[2]+X[4]+X[6])/4.0;
+                    double y1 = (Y[0]+Y[2]+Y[4]+Y[6])/4.0;
+
+                    double dx = sqrt(pow((x2-x1), 2.0) + pow((y2-y1), 2.0) );
+
+                    // calculate dy
+
+                    x2 = (X[2]+X[3]+X[6]+X[7])/4.0;
+                    y2 = (Y[2]+Y[3]+Y[6]+Y[7])/4.0;
+
+                    x1 = (X[0]+X[1]+X[4]+X[5])/4.0;
+                    y1 = (Y[0]+Y[1]+Y[4]+Y[5])/4.0;
+
+                    double dy = sqrt(pow((x2-x1), 2.0) + pow((y2-y1), 2.0));
+
+                    // calculate dz
+
+                    double z2 = (Z[4]+Z[5]+Z[6]+Z[7])/4.0;
+                    double z1 = (Z[0]+Z[1]+Z[2]+Z[3])/4.0;
+
+                    double dz = z2-z1;
+
+                    m_dx[n] = dx;
+                    m_dy[n] = dy;
+                    m_dz[n] = dz;
+
+                    m_depth[n] =  (z2 + z1) / 2.0;
+
+                    n++;
+                }
+            }
+        }
+    }
+
+    void EclipseGrid::getCellCorners(const std::array<int, 3>& ijk, const std::array<int, 3>& dims,
+                                     std::vector<double>& X,
+                                     std::vector<double>& Y,
+                                     std::vector<double>& Z) const
+    {
+
+        std::array<int, 8> zind;
+        std::array<int, 4> pind;
+        
+        if (X.size() < 8 || Y.size() < 8 || Z.size() < 8) {
```

Comment:
I guess this should be `!= 8` and not `< 8`?

Reply:
Changed to std::array<double,8> assuming that I can delete the if block. correct ?

Reply:
Yes - delete the `if`